pkgs/by-name/fr/frequest/package.nix

```diff
@@ -0,0 +1,69 @@
+{ lib, stdenv, fetchFromGitHub, qt5 }:
+
+let
+  version = "1.2a";
+
+  src = fetchFromGitHub {
+    owner = "fabiobento512";
+    repo = "FRequest";
+    rev = "v${version}";
+    hash = "sha256-fdn3MK5GWBOhJjpMtRaytO9EsVzz6KJknDhqWtAyXCc=";
+  };
+
+  commonLibsSrc = fetchFromGitHub {
+    owner = "fabiobento512";
+    name = "CommonLibs";
+    repo = "CommonLibs";
+    rev = "d3906931bb06ddf4194ff711a59e1dcff80fa82f";
+    hash = "sha256-iLJJ95yJ+VjNPuk8fNEDvYBI0db0rcfJF12a9azGv1Y=";
+  };
+
+  commonUtilsSrc = fetchFromGitHub {
+    owner = "fabiobento512";
+    name = "CommonUtils";
+    repo = "CommonUtils";
+    rev = "53970984f6538d78350be1b9426032bcb5bcf818";
+    hash = "sha256-nRv9DriSOuAiWhy+KkOVNEz5oSgNNNJZqk8sNwgbx8U=";
+  };
+
+in stdenv.mkDerivation {
+
+  pname = "frequest";
+  inherit version;
+
+  srcs = [
+    src
+    commonLibsSrc
+    commonUtilsSrc
+  ];
+  sourceRoot = src.name;
+
+  buildInputs = [
+    qt5.qtbase
+  ];
+
+  nativeBuildInputs = [
+    qt5.wrapQtAppsHook
+    qt5.qmake
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D FRequest $out/bin/FRequest
+    install -D LinuxAppImageDeployment/frequest.desktop $out/share/applications/frequest.desktop
+    install -D LinuxAppImageDeployment/frequest_icon.png $out/share/icons/hicolor/128x128/apps/frequest_icon.png
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "A fast, lightweight and opensource desktop application to make HTTP(s) requests";
+    homepage = "https://fabiobento512.github.io/FRequest";
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ eliandoran ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
Please try to enable all platforms..

Reply:
@doronbehar , it's a very good point. Usually when I package something, I package it for the platform I use the most. I don't feel comfortable adding more platforms unless I actually look into them because of the small things that might not work correctly (for example runtime errors on different platforms).

Seems that it _kind of_ works on Mac, but it expects a mutable directory in which to work. I wonder why it doesn't fail as such on Linux, there it seems to take the current directory instead.

I'd probably need to write some kind of wrapper script for darwin. 🤔 

<img width="724" alt="image" src="https://github.com/NixOS/nixpkgs/assets/21236836/0703b122-f19f-46ba-a3f7-f7540b47f338">

Reply:
@doronbehar , I would prefer not to block the merging of the PR until we manage to get darwin support.
I'll most likely have to see with the upstream developer if we can add some kind of environment variable to change the path to the configuration files, or simply apply some patch. I think `~/.config/frequest` would be a more suitable location for the configuration files than the current working directory.